### DISCUSSION TITLE
Spike in ram usage when using fig.construct_update_data in a dash app

### DISCUSSION THREAD

**Discussion by surya-uv** [predict-idlab/plotly-resampler#313] [Q&A]
Hi,

I observed there is a spike in RAM usage when I zoom on a figure with subplot.
For context, I am using the resampler in a dash app callback function with trace updater. where replay out data is the trigger for the callback.
But when I use the example code provided in the plotly-resampler documentation. there is no change in RAM usage.
Are these ram spikes normal on dash app? 
Also, Is there an example code which can show how to implement resampler in dash app with callbacks?

Thanks in advance

**Comment by darshanagopal** [comment]
RAM usage spikes when zooming in on a figure with subplots can sometimes occur in Dash apps, especially if you're dealing with large datasets or complex visualizations. The resampling process may demand more memory during interactions like zooming, as it might be recalculating or fetching additional data points to provide a more detailed view.

However, the fact that you don’t see this behavior when using the example code from the Plotly-Resampler documentation suggests that there might be something specific to your implementation or dataset that's causing the spike. It might be related to how the data is being handled or the specific way subplots are managed in your callback.
